### PR TITLE
install role: improved determination of latest GitHub release

### PR DESCRIPTION
**Pull request by felixfontein**
Nightly CI is failing a lot recently due to GH API rate-limiting.

### CODE REVIEW

File: roles/install/tasks/github.yml
Comment (kristianheljas) [inline roles/install/tasks/github.yml:38]: Relying on undocumented HTTP redirect seems quite fragile and might not avoid preleases and drafts as the original one does.

Have you thought of using [automatically provided `GITHUB_TOKEN` secret](https://docs.github.com/en/actions/security-guides/automatic-token-authentication) and passing that to the role, increasing hourly limit from 60 to 1000?

Comment (felixfontein) [reply roles/install/tasks/github.yml:38]: I don't think the redirect is really undocumented. https://docs.github.com/en/repositories/releasing-projects-on-github/linking-to-releases explicitly mentions /releases/latest (though it does not mention it is a redirect).

**Comment by felixfontein** [comment]
Hmm, if I want to pass something passed in to ansible-test to a role argument spec, I get `"argument 'sops_github_token' is of type <class 'ansible.parsing.yaml.objects.AnsibleUnicode'> and we were unable to convert to string: 'NoneType' object is not callable"`, and if I use the Jinja `| string` filter to try to avoid this, I get `"argument 'sops_github_token' is of type <class 'ansible.utils.native_jinja.NativeJinjaText'> and we were unable to convert to string: 'NoneType' object is not callable"`. Looks like a bug in ansible-core that needs fixing :-)

**Comment by felixfontein** [comment]
I've removed the GitHub token support for now, I think it's best to first continue with the original way and try the token in an additional PR (and wait until that bug is fixed in ansible-core before proceeding).

**Review by kristianheljas — APPROVED** [review]
@felixfontein Thank you for considering the token option, but I agree that getting nightly CI working takes precedence here since passing the token is troublesome.

If the /releases/latest redirect would change at some point it will fall back to the API anyway - your regex is sufficiently specific :)

**Comment by felixfontein** [comment]
@kristianheljas @russoz thanks for reviewing this!

**Comment by felixfontein** [comment]
I created #132 for the GH token support.

**Comment by kristianheljas** [comment]
@felixfontein Once #132 is merged, would it make sense to undo this http redirect check? That way prerelease and draft releases would be excluded again.

**Comment by felixfontein** [comment]
@kristianheljas I don't see how the redirect is related to prerelease or draft releases. `latest` never points to either of them.

**Comment by kristianheljas** [comment]
Ah, I stand corrected - I thought "latest" just pointed to the literally latest release on the releases page but it doesn't, indeed.

**Comment by felixfontein** [comment]
`latest` is basically what the GH repository maintainer decides should be labelled as `latest`, the UI only ensures that drafts and prereleases won't be `latest`.

I'm currently thinking of making the behavior configurable (with default to use the API call), but I'm not 100% sure yet.

**Comment by felixfontein** [comment]
I've implemented this in #133.

**Pull request merged**
